packages/@ember/-internals/glimmer/tests/integration/application/debug-render-tree-test.ts

```diff
@@ -4,6 +4,8 @@ import {
   moduleFor,
   strip,
 } from 'internal-test-helpers';
+import { ExpectDeprecationFunc } from '../../../../../../internal-test-helpers/lib/ember-dev/deprecation';
+declare let expectDeprecation: ExpectDeprecationFunc;
```

Comment:
we probably should declare this globally somewhere, rather than just in the same file. It can probably be done in the same file that `ExpectDeprecationFunc` is defined

Reply:
Thanks for the feedback 👍

I'm still trying to get my head around TypeScript. I see there's already a global declaration for `expectDeprecation` and some functions injected into `window`
 https://github.com/emberjs/ember.js/blob/0fa78e0dd84a1c8aa572c2db9bf12664602248bf/packages/internal-test-helpers/lib/ember-dev/deprecation.ts#L15-L22

https://github.com/emberjs/ember.js/blob/0fa78e0dd84a1c8aa572c2db9bf12664602248bf/packages/internal-test-helpers/lib/ember-dev/deprecation.ts#L80

Should I be exporting the `expectDeprecation` function instead of having it created it inline, or is there some other magic I can rely on?

Reply:
@pzuraq @rwjblue I'm struggling a bit to figure out how to get the `declare let expectDeprecation: ExpectDeprecationFunc;` typing available from `internal-test-helpers`. My TypeScript foo isn't great.

There was an attempt in [`a7f16ce` (#19407)](https://github.com/emberjs/ember.js/pull/19407/commits/a7f16ce2995376248cbbc155ca69cb928f7a4008) but unfortunately the value is `undefined` which makes as I amn't assigning it to anything. Also it's not straight-forward to export the existing `expectDeprecation` as it's a method on the `DeprecationAssert` and accesses instance methods on the class.

Should the `declare let` be added to the `internal-helpers/index.d.ts` file instead?

Reply:
Discussed offline, we can export the `expectDeprecation` function in the same file where it is defined as a `let`:

```ts
export let expectDeprecation: ExpectDeprecationFunc = () => {
  throw new Error('err');
};
```

It can then be assigned when we inject the deprecation functions

Reply:
@pzuraq This should be ready to go now